RecoTracker/LSTCore/src/alpaka/NeuralNetwork.h

```diff
@@ -39,102 +39,88 @@ namespace lst::t5dnn {
-    float x5 = xVec[4];
-    // Unpack y-coordinates of hits
-    float y1 = yVec[0];
-    float y2 = yVec[1];
-    float y3 = yVec[2];
-    float y4 = yVec[3];
-    float y5 = yVec[4];
+    // Constants
+    constexpr unsigned int kinputFeatures = 18;
+    constexpr unsigned int khiddenFeatures = 32;
+
     // Unpack module indices
     unsigned int mdIndex1 = mdIndices[0];
     unsigned int mdIndex2 = mdIndices[1];
     unsigned int mdIndex3 = mdIndices[2];
     unsigned int mdIndex4 = mdIndices[3];
     unsigned int mdIndex5 = mdIndices[4];
 
-    // Compute some convenience variables
-    short layer2_adjustment = 0;
-    if (modulesInGPU.layers[lowerModuleIndices[0]] == 1) {
-      layer2_adjustment = 1;  // get upper segment to be in second layer
-    }
-    unsigned int md_idx_for_t5_eta_phi =
-        segmentsInGPU.mdIndices[2 * tripletsInGPU.segmentIndices[2 * innerTripletIndex + layer2_adjustment]];
+    float eta1 = alpaka::math::abs(acc, mdsInGPU.anchorEta[mdIndex1]);  // inner T3 anchor hit 1 eta (t3_0_eta)
+    float eta2 = alpaka::math::abs(acc, mdsInGPU.anchorEta[mdIndex2]);  // inner T3 anchor hit 2 eta (t3_2_eta)
+    float eta3 = alpaka::math::abs(acc, mdsInGPU.anchorEta[mdIndex3]);  // inner T3 anchor hit 3 eta (t3_4_eta)
+    float eta4 = alpaka::math::abs(acc, mdsInGPU.anchorEta[mdIndex4]);  // outer T3 anchor hit 4 eta (t3_2_eta)
+    float eta5 = alpaka::math::abs(acc, mdsInGPU.anchorEta[mdIndex5]);  // outer T3 anchor hit 5 eta (t3_4_eta)
+
+    float z1 = alpaka::math::abs(acc, mdsInGPU.anchorZ[mdIndex1]);  // inner T3 anchor hit 1 z (t3_0_z)
+    float z2 = alpaka::math::abs(acc, mdsInGPU.anchorZ[mdIndex2]);  // inner T3 anchor hit 2 z (t3_2_z)
+    float z3 = alpaka::math::abs(acc, mdsInGPU.anchorZ[mdIndex3]);  // inner T3 anchor hit 3 z (t3_4_z)
+    float z4 = alpaka::math::abs(acc, mdsInGPU.anchorZ[mdIndex4]);  // outer T3 anchor hit 4 z (t3_2_z)
+    float z5 = alpaka::math::abs(acc, mdsInGPU.anchorZ[mdIndex5]);  // outer T3 anchor hit 5 z (t3_4_z)
+
+    float r1 = mdsInGPU.anchorRt[mdIndex1];  // inner T3 anchor hit 1 r (t3_0_r)
+    float r2 = mdsInGPU.anchorRt[mdIndex2];  // inner T3 anchor hit 2 r (t3_2_r)
+    float r3 = mdsInGPU.anchorRt[mdIndex3];  // inner T3 anchor hit 3 r (t3_4_r)
+    float r4 = mdsInGPU.anchorRt[mdIndex4];  // outer T3 anchor hit 4 r (t3_2_r)
+    float r5 = mdsInGPU.anchorRt[mdIndex5];  // outer T3 anchor hit 5 r (t3_4_r)
+
+    // Build the input feature vector using pairwise differences after the first hit
+    float x[kinputFeatures] = {
+        eta1 / kEta_norm,  // inner T3: First hit eta normalized
+        z1 / kZ_max,       // inner T3: First hit z normalized
+        r1 / kR_max,       // inner T3: First hit r normalized
 
-    float t5_eta = mdsInGPU.anchorEta[md_idx_for_t5_eta_phi];
+        eta2 - eta1,         // inner T3: Difference in eta between hit 2 and 1
+        (z2 - z1) / kZ_max,  // inner T3: Difference in z between hit 2 and 1 normalized
```

Comment:
here and below, are we losing performance around eta or z around zero by taking deltas of abs values? Perhaps deltas of signed values are more useful?

Reply:
I just gave this a try but didn't see any improvement. Marking as resolved.